lib/ui/window/pointer_data_packet_converter.cc

```diff
@@ -74,7 +74,42 @@ void PointerDataPacketConverter::ConvertPointerData(
         break;
       }
       case PointerData::Change::kAdd: {
-        FML_DCHECK(states_.find(pointer_data.device) == states_.end());
+        auto iter = states_.find(pointer_data.device);
+        if (iter != states_.end()) {
+          // Synthesizes a remove event if the pointer was not previously
+          // removed.
+          PointerState state = iter->second;
+          PointerData synthesized_data = pointer_data;
+          synthesized_data.physical_x = state.physical_x;
+          synthesized_data.physical_y = state.physical_y;
+          synthesized_data.pan_x = state.pan_x;
+          synthesized_data.pan_y = state.pan_y;
+          synthesized_data.scale = state.scale;
+          synthesized_data.rotation = state.rotation;
+          synthesized_data.buttons = state.buttons;
+
+          if (delegate_.ViewExists(state.view_id)) {
```

Comment:
Your comment in the PR description is excellent, consider adding it here to explain what's happening:

```suggestion
          // The framework expects an "add" event to always follow a "remove" event,
          // and "remove" events with invalid views are ignored.
          // To meet the framework's expectations, the view ID of the "add" event
          // will be used for the "remove" event if the old view has been destroyed.
          if (delegate_.ViewExists(state.view_id)) {
```

Reply:
Done